mobile/lib/features/student_management/data/data_sources/student_local_data_source.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:mobile/core/data/database.dart';
+import 'package:mobile/core/error/exceptions.dart';
+import 'package:moor/ffi.dart';
+import 'package:moor/moor.dart';
+
+abstract class StudentLocalDataSource {
+  /// Gets the cached list of [Student].
+  ///
+  /// Returns an empty list if no [Student] is cached.
+  ///
+  /// Throws [CacheException] if something wrong happens.
+  Future<List<StudentModel>> getStudentsFromCache(ClassroomModel clasroomModel);
+
+  /// Deletes the [Student] passed.
+  ///
+  /// Throws [CacheException] if the [Student] is not cached.
+  Future<void> deleteStudentFromCache(StudentModel studentModel);
+
+  /// Caches the new [Student] passed.
+  ///
+  /// Throws [CacheException] if something wrong happens.
+  Future<StudentModel> cacheNewStudent(StudentModel studentModel);
+
+  /// Updated in cache the [Student] passed.
+  ///
+  /// Throws [CacheException] if [Student] is not cached.
+  Future<StudentModel> updateCachedStudent(StudentModel studentModel);
+}
+
+class StudentLocalDataSourceImpl implements StudentLocalDataSource {
+  final Database database;
+
+  StudentLocalDataSourceImpl({@required this.database});
+
+  @override
+  Future<StudentModel> cacheNewStudent(StudentModel studentModel) async {
+    try {
+      bool nullToAbsent = true;
+      final studentCompanion = studentModel.toCompanion(nullToAbsent);
+      final studentPk = await this.database.insertStudent(studentCompanion);
+      return StudentModel(
+          localId: studentPk,
+          firstName: studentModel.firstName,
+          lastName: studentModel.lastName,
+          classroomId: studentModel.classroomId);
+    } on SqliteException {
+      throw CacheException();
+    }
+  }
+
+  @override
+  Future<void> deleteStudentFromCache(StudentModel studentModel) async {
+    var pk = studentModel.localId;
+    try {
+      var done = await this.database.deleteStudent(pk);
+      if (done != 1) {
+        throw SqliteException(787, "The table don't have this entry");
```

Comment:
Typo

Reply:
don't get it

Reply:
The table doesn't have this entry

Reply:
> WTF a just changed from **don't** to **doesn't** and udpate `database.dart` now coverage isn't passing

Probably the database.dart, with one new line